### DISCUSSION TITLE
Can't login to Community Edition

### DISCUSSION THREAD

**Discussion by drumwhizz** [shellhub-io/shellhub#2341] [Q&A] [answered]
I have successfully followed the setup instructions for deploying the self hosted shellhub service (v0.10.9) locally on my Ubuntu desktop. I have created my first user and namespace, but every time I try and login to the service (with my credentials) I get a small dialog popping up saying "The request has failed. Please try again". It does this even if I type in rubbish into the username and password.

I have had a look in the logs but are none the wiser as to what to look for.

**Comment by drumwhizz** [comment]
phil@phil-desktop:~/shellhub$ cat docker-compose.override.yml 
version: '3.7'
services:
  mongo:
    volumes:
      - /usr/local/mongo/data:/data/db

**Comment by henrybarreto** [comment]
Hello, Phil!

Thank you for reporting it. I'm working on this issue now.

Could you check your user on MongoDB database? 

```
 ./bin/docker-compose exec mongo mongo main --eval "db.users.find()"
```

If it's okay, could you share the request logs what have happened?

**Reply by drumwhizz** [reply]
Request Logs?

**Reply by henrybarreto** [reply]
My idea is a bit manual. 

Please open the API logs, try to log in, and post here the API logs.

```
./bin/docker-compose logs -f api
```

**Reply by drumwhizz** [reply]
Logging in is not generating any log entries. All I see is this

api_1      | time="2023-01-05T13:47:44Z" level=fatal msg="Failed to initialize Sentry" error="[Sentry] DsnParseError: invalid scheme"
api_1      | time="2023-01-05T13:48:45Z" level=info msg="Setting log level" log_level=info
api_1      | time="2023-01-05T13:48:45Z" level=info msg="Starting Sentry"
api_1      | time="2023-01-05T13:48:45Z" level=fatal msg="Failed to initialize Sentry" error="[Sentry] DsnParseError: invalid scheme"
api_1      | time="2023-01-05T13:49:45Z" level=info msg="Setting log level" log_level=info
api_1      | time="2023-01-05T13:49:45Z" level=info msg="Starting Sentry"
api_1      | time="2023-01-05T13:49:45Z" level=fatal msg="Failed to initialize Sentry" error="[Sentry] DsnParseError: invalid scheme"
api_1      | time="2023-01-05T13:50:46Z" level=info msg="Setting log level" log_level=info
api_1      | time="2023-01-05T13:50:46Z" level=info msg="Starting Sentry"
api_1      | time="2023-01-05T13:50:46Z" level=fatal msg="Failed to initialize Sentry" error="[Sentry] DsnParseError: invalid scheme"
api_1      | time="2023-01-05T13:51:46Z" level=info msg="Setting log level" log_level=info
api_1      | time="2023-01-05T13:51:46Z" level=info msg="Starting Sentry"
api_1      | time="2023-01-05T13:51:46Z" level=fatal msg="Failed to initialize Sentry" error="[Sentry] DsnParseError: invalid scheme"

Sentry failing to initialise

**Reply by henrybarreto** [reply]
When sentry DSN environmental variable has value, `SHELLHUB_SENTRY_DSN`, the server will try to initialize it. If something happened during this initialization, the server will not start.

 Could you disable this feature to check it out?

**Reply by drumwhizz** [reply]
Is that done in .env.override? What do I need to set it?

**Reply by henrybarreto** [reply]
By default, it is disabled. You can verify it, looking for the last line on `.env` file, or create a file called `.env.override` and set it:

```
SHELLHUB_SENTRY_DSN=""
```

**Reply by drumwhizz** [reply]
Its disabled by default:

`# Sentry DSN`
`SHELLHUB_SENTRY_DSN=""`

**Reply by drumwhizz** [reply]
Have just restarted and I am still seeing those errors

`api_1      | time="2023-01-05T14:36:05Z" level=info msg="Setting log level" log_level=info`
`api_1      | time="2023-01-05T14:36:05Z" level=info msg="Starting Sentry"`
`api_1      | time="2023-01-05T14:36:05Z" level=fatal msg="Failed to initialize Sentry" error="[Sentry] DsnParseError: invalid scheme"`

**Reply by henrybarreto** [reply]
Could you stop everything and start again instead of restart? I believe it will load the env variables and fix it.

```
./bin/docker-compose stop && ./bin/docker-compose up
```

**Reply by drumwhizz** [reply]
Here is the log..
[shellhub-log.txt](https://github.com/shellhub-io/shellhub/files/10353372/shellhub-log.txt)

**Reply by henrybarreto** [reply]
Okay, it is a bit strange...

Could you post here the API container envs?

```
./bin/docker-compose exec api env
```

**Reply by drumwhizz** [reply]
`phil@phil-desktop:~/shellhub$ ./bin/docker-compose exec api env`
`INFO: Loading ./.env.override`
`Error response from daemon: Container b31e3965f1ae4f66b8d5657fea26c77ecf546aa98a4ca5ad6530cdbf89ae6ce4 is restarting, wait until the container is running`

**Reply by drumwhizz** [reply]
The api is stuck in a restart cycle

api_1      | time="2023-01-06T08:01:03Z" level=info msg="Setting log level" log_level=info
api_1      | time="2023-01-06T08:01:03Z" level=info msg="Starting Sentry"
api_1      | time="2023-01-06T08:01:03Z" level=fatal msg="Failed to initialize Sentry" error="[Sentry] DsnParseError: invalid scheme"
api_1      | time="2023-01-06T08:01:29Z" level=info msg="Setting log level" log_level=info
api_1      | time="2023-01-06T08:01:29Z" level=info msg="Starting Sentry"
api_1      | time="2023-01-06T08:01:29Z" level=fatal msg="Failed to initialize Sentry" error="[Sentry] DsnParseError: invalid scheme"
api_1      | time="2023-01-06T08:02:21Z" level=info msg="Setting log level" log_level=info
api_1      | time="2023-01-06T08:02:21Z" level=info msg="Starting Sentry"
api_1      | time="2023-01-06T08:02:21Z" level=fatal msg="Failed to initialize Sentry" error="[Sentry] DsnParseError: invalid scheme"
api_1      | time="2023-01-06T08:03:21Z" level=info msg="Setting log level" log_level=info
api_1      | time="2023-01-06T08:03:21Z" level=info msg="Starting Sentry"
api_1      | time="2023-01-06T08:03:21Z" level=fatal msg="Failed to initialize Sentry" error="[Sentry] DsnParseError: invalid scheme"

**Comment by drumwhizz** [comment]
phil@phil-desktop:/shellhub$ ./bin/docker-compose exec mongo mongo main --eval "db.users.find()"
MongoDB shell version v4.4.8
connecting to: mongodb://127.0.0.1:27017/main?compressors=disabled&gssapiServiceName=mongodb
Implicit session: session { "id" : UUID("99ca9ef1-ece9-41af-a714-4b89d3e3a4ef") }
MongoDB server version: 4.4.8
{ "_id" : ObjectId("63b69fee386663b53c645f10"), "max_namespaces" : -1, "confirmed" : true, "created_at" : ISODate("2023-01-05T10:01:18.058Z"), "last_login" : ISODate("0001-01-01T00:00:00Z"), "name" : "phil", "email" : "phil@8power.com", "username" : "phil", "password" : "2bb80d537b1da3e38bd30361aa855686bde0eacd7162fef6a25fe97bf527a25b", "namespaces" : 3 }
{ "_id" : ObjectId("63b6afbf053bdac67f8cf236"), "max_namespaces" : -1, "confirmed" : true, "created_at" : ISODate("2023-01-05T11:08:47.364Z"), "last_login" : ISODate("0001-01-01T00:00:00Z"), "name" : "phil", "email" : "phil@8power.com", "username" : "phil", "password" : "2bb80d537b1da3e38bd30361aa855686bde0eacd7162fef6a25fe97bf527a25b" }
{ "_id" : ObjectId("63b6b51fa1dc220626a2a9f4"), "max_namespaces" : -1, "confirmed" : true, "created_at" : ISODate("2023-01-05T11:31:43.669Z"), "last_login" : ISODate("0001-01-01T00:00:00Z"), "name" : "phil", "email" : "phil@8power.com", "username" : "phil", "password" : "2bb80d537b1da3e38bd30361aa855686bde0eacd7162fef6a25fe97bf527a25b" }
phil@phil-desktop:~/shellhub$

**Reply by otavio** [reply]
@henrybarreto and I are looking at this, and we are wondering how you ended with **three** users with the same email. Below is a snippet of your response:

```json
{ ..."created_at" : ISODate("2023-01-05T10:01:18.058Z"), ..., "namespaces" : 3 }
{ ..."created_at" : ISODate("2023-01-05T11:08:47.364Z"), ... }
{ ..."created_at" : ISODate("2023-01-05T11:31:43.669Z"), ... }
```
So we'd like to know how you did the installation in detail. You are using MongoDB with an external storage directory for persistence. Have you reused an existing database from an older tag, or did you start the installation directly inside the 0.10.9 tag?

**Reply by drumwhizz** [reply]
I restarted the container and repeated the processes of configuring the database 3 times. I could not see any files in the host DB directory and assumed that mongo writes were not working.

**Reply by drumwhizz** [reply]
WRT, the database, my recollection is that I pulled the tip of the repo and attempted to build it but it failed due to the fact you have made breaking changes with the new_ui. I then checked out the v0.10.9 tag and restarted the service successfully.

**Comment by henrybarreto** [comment]
@drumwhizz, I have reproduced that problem locally. I'm working directly on it now.

**Comment by henrybarreto** [comment, marked as answer]
@drumwhizz, the problem seems to be with the _docker-compose_ version what you are using.

> I'm assuming you are using the _Ubuntu 20.04_, sorry.

Looking for the _docker-compose_ package on your distro, I've checked that this version, **[1.25.0](https://packages.ubuntu.com/focal/docker-compose)**, doesn't support empty (`""`) strings as null on `.env`.

To work around that issue, you can run that script. It'll remove all `""` from the `.env`, allowing the API server to run normally. 
```
sed -i 's,"",,g' .env
```

**Reply by drumwhizz** [reply]
I have just tried this and it has made no difference to the behaviour.

**Reply by henrybarreto** [reply]
@drumwhizz, could you access our gitter for a better conversation?

https://gitter.im/shellhub-io/community
